### PR TITLE
[CI] sign artifacts for the release (partially implemented)

### PR DESCRIPTION
**Pull request by v1v**
## What

Signing process is decoupled since it's one of the responsibilities of the unified release process, therefore this particular approach will allow us to use the `internal-ci` process and trigger the signing process.

For such, it's required to create another Multibranch Pipeline in the `internal-ci` instance, and the release process will be driving when a tag release is created.

## Why

Normalise the process of any release that generates specific binaries for specific platforms. Therefore our uses will be able to benefit from the same signing validations as we do for other products.

## Important

This PR is partially implemented since it's required to be tested in the `internal-ci`, how do I wanna test this out?

- Merge this PR
- Create tag `0.0.1-test`
- Pipeline in the `internal-ci` should be triggered.
- If everything goes smooth then:
  - a) I'll poke @mgreau regarding the signing artifacts pipeline
  - b) Validate the notifications by slack or email fi required.
  - c) Create a PR with the above changes/suggestions

## Tasks

- [x] Review bucket
- [x] Review credentials
- [x] Review notifications
- [x] Create JJB in the `internal-ci` only for tags creation with the pattern `v[0-9]+.[0-9]+*`
- [x] Signing artifact component requirements
- [x] Enable release step https://github.com/elastic/apm-agent-php/pull/172/files#r493708279

## Issues

Caused by https://github.com/elastic/apm-agent-php/issues/117

## Workflow

- Pre release slack notification

image

- `approve` link

image

- Blue Ocean view

image

- Release slack notification

image

- Signing component call

image

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -293,7 +325,9 @@ pipeline {
   }
   post {
     cleanup {
-      notifyBuildResult()
+      whenFalse(env.JENKINS_URL.contains('internal-ci')) {
```
Comment (v1v) [inline .ci/Jenkinsfile:328]: We don't need this particular post-build stage in the internal-ci instance

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -237,14 +237,18 @@ pipeline {
...
       when {
         beforeAgent true
-        tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+        allOf {
+          tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+          expression { env.JENKINS_URL.contains('internal-ci') }
+        }
       }
       stages {
         stage('Notify') {
```
Comment (v1v) [inline .ci/Jenkinsfile:254]: This stage might be required to be evaluated as we need to review if we got permissions to interact with the input parameters in the UI

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -237,14 +237,18 @@ pipeline {
...
       options {
         skipDefaultCheckout()
         timeout(time: 12, unit: 'HOURS')
       }
       when {
         beforeAgent true
-        tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+        allOf {
+          tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+          expression { env.JENKINS_URL.contains('internal-ci') }
```
Comment (v1v) [inline .ci/Jenkinsfile:250]: https://github.com/elastic/apm-pipeline-library/pull/727

```suggestion
          expression { isInternalCI() }
```

**Review by SergeyKleyman — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 4]
```diff
@@ -261,6 +261,7 @@ pipeline {
         // For testing purposes only
         TAG_NAME = 'test'
       }
+      agent { label 'linux && docker && ubuntu-18.04 && immutable' }
```
Comment (v1v) [inline .ci/Jenkinsfile:265]: Fixed with https://github.com/elastic/apm-agent-php/pull/186

File: .ci/Jenkinsfile [hunk 5]
```diff
@@ -245,27 +245,63 @@ pipeline {
...
+                  sharedPublicly: false,
+                  showInline: true)
+              echo "build(wait: true, propagate: true, job: '<TBD>', parameters: [string(name: 'URL', value: ${env.BUCKET_PATH})])"
+              dir('signed') {
+                googleStorageDownload(bucketUri: "${env.BUCKET_PATH}/*",
+                    credentialsId: env.BUCKET_CREDENTIALS,
+                    localDirectory: 'build/packages/',
+                    pathPrefix: "${BUCKET_SUBFOLDER}")
+                stash allowEmpty: false, name: 'signed-package', useDefaultExcludes: false
+              }
```
Comment (v1v) [inline .ci/Jenkinsfile:304]: @mgreau , this is the section I coded to support the artifact-signing component, how does it work?

- generated artifacts are uploaded to the location `gs://internal-ci-artifacts/apm-agent-php/<tag>` where `<tag>` follow the format `v[0-9]+.*`

For instance, I was using only for testing purposes TAG=test and files were uploaded

image

- Then L297 calls the build step (so far an echo), I need your input here what's the parameter name and expected format besides where those signed packages will be stored in the bucket.

File: .ci/Jenkinsfile [hunk 6]
```diff
@@ -275,19 +311,26 @@ pipeline {
...
           steps {
             deleteDir()
             unstash 'source'
             dir("${BASE_DIR}") {
-              unstash 'package'
+              unstash 'signed-package'
               withCredentials([string(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7', variable: 'GITHUB_TOKEN')]) {
-                sh script: 'make -f .ci/Makefile release', label: 'release'
+                echo "SKIPPED: for testing purposes"
+                // sh script: 'make -f .ci/Makefile release', label: 'release'
```
Comment (v1v) [inline .ci/Jenkinsfile:325]:
```suggestion
                sh script: 'make -f .ci/Makefile release', label: 'release'
```

**Pull request merged**